### PR TITLE
ipdef: Simplify and rename the ip package

### PR DESCRIPTION
**Pull request by ejj**
The `ip` package had a lot of unnecessary library code that would have
been more clearly implemented by its callers directly.  Also, it's
name, `ip`, was rather annoying as it prevented the programmer from
using `ip` as a variable name.  This patch solves the problem by
creating a new package, `ipdef`, which contains our network constants
without all of the helper functions in the old `ip` package.

### CODE REVIEW

File: minion/etcd/minion.go [hunk 1]
```diff
@@ -252,21 +245,25 @@ func syncSubnet(conn db.Conn, store Store, minion db.Minion) {
...
+	return ipNet
 }
 
-func subnetKey(subnet string) string {
-	return path.Join(subnetStore, subnet)
+func subnetKey(subnet net.IPNet) string {
+	return path.Join(subnetStore, subnet.IP.String())
 }
+
+var rand32 = rand.Uint32
```
Comment (mmussomele) [inline minion/etcd/minion.go]: would this be better at the top?

File: minion/etcd/minion_test.go [hunk 2]
```diff
@@ -130,23 +131,18 @@ func TestFilter(t *testing.T) {
...
 	store := newTestMock()
-	subnetAttempts = 5000 // big so we don't error unless something is wrong
-	defer func() {
-		subnetAttempts = 1000
-	}()
-	minionMaskBits, _ := ip.SubMask.Size()
+	minionMaskBits, _ := ipdef.SubMask.Size()
 	emptyBits := uint32(0xffffffff >> uint(minionMaskBits))
 	for i := 0; i < 3500; i++ {
 		sub, err := generateSubnet(store, db.Minion{PrivateIP: "10.1.0.1"})
```
Comment (mmussomele) [inline minion/etcd/minion_test.go]: No reason not to call this `subnet` now, might help with clarity slightly

File: minion/network/plugin/plugin.go [hunk 3]
```diff
@@ -199,3 +200,5 @@ func endpointExists(eid string) (bool, error) {
 	_, name := network.VethPairNames(eid)
 	return network.LinkExists("", name)
 }
+
+var rand32 = rand.Uint32
```
Comment (mmussomele) [inline minion/network/plugin/plugin.go]: same here with moving to the top

File: minion/etcd/minion.go (hunk 1 above)
Comment (ejj) [inline minion/etcd/minion.go]: So here's my reasoning on this, i generally like functions to be ordered from most important to least important, with variables/constants at the top.  So if you consider this a function, it's the least important and thus belongs at the bottom.  It's also similar to some other packages where we have the sorting interface near the bottom.  Basically, in short, readers shouldn't care about it hence its location.

That said, I don't feel that strongly about it...

Comment (mmussomele) [inline minion/etcd/minion.go]: Yea sure then. I don't particularly care either, seems like you've thought about it though so that's fine with me

**Pull request merged**